### PR TITLE
[PM-10877] New workflows for version calculation and workflow dispatch

### PR DESCRIPTION
**Pull request by vvolkgang**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10877

## 📔 Objective

First step in moving version calculation to it's own file. Also adds a new composite action to dispatch, wait and download artifacts from other workflows. We need these in `main` before we're able to use them in our existing build related workflows.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/_version.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+
+          if [[ "$current_year" == "$latest_major_version" && "$current_month" == "$latest_minor_version" ]]; then
+            version_name="${latest_major_version}.${latest_minor_version}.$(($latest_patch_version + 1))"
+          else
+            version_name="${current_year}.${current_month}.0"
+          fi
+
+          output "$version_name"
+
```
Comment (KatherineInCode) [inline .github/workflows/_version.yml:90]: ⛏️ Double newline

File: .github/workflows/_version.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      - name: Log inputs to job summary
+        run: |
+          echo "<details><summary>Workflow Inputs</summary>" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo '```json' >> $GITHUB_STEP_SUMMARY
+          echo '${{ toJson(inputs) }}' >> $GITHUB_STEP_SUMMARY
+          echo '```' >> $GITHUB_STEP_SUMMARY
+          echo "</details>" >> $GITHUB_STEP_SUMMARY
+
+      - name: echo distinct ID ${{ github.event.inputs.distinct_id }}
```
Comment (KatherineInCode) [inline .github/workflows/_version.yml:38]: ⛏️ Capitalize

File: .github/workflows/_version.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        run: echo ${{ github.event.inputs.distinct_id }}
+
+      - name: Checkout repository
```
Comment (KatherineInCode) [inline .github/workflows/_version.yml:41]: ⛏️ 
```suggestion
      - name: Check out repository
```

File: .github/workflows/_version.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
+---
+name: Calculate Version Name and Number
+
+
+on:
+  workflow_dispatch:
+    inputs:
+      base_version_number:
```
Comment (KatherineInCode) [inline .github/workflows/_version.yml:8]: 👍🏻 I like this!

**Review by fedemkr — APPROVED** [review]

**Review by KatherineInCode — APPROVED** [review]

**Pull request merged**